R/merge.R

```diff
@@ -82,8 +82,9 @@ merge.data.table = function(x, y, by = NULL, by.x = NULL, by.y = NULL, all = FAL
       othercolsx = setdiff(nm_x, by)
       if (length(othercolsx)) {
         tmp = rep.int(NA_integer_, length(missingyidx))
-        # TO DO: use set() here instead..
-        yy = cbind(yy, x[tmp, othercolsx, with = FALSE])
+        nx = c(names(yy), names(x[tmp, othercolsx, with = FALSE]))
+        nx = make.unique(nx)
+        set(yy, NULL, tail(nx, -ncol(yy)), x[tmp, othercolsx, with = FALSE])
```

Comment:
is there a noticeable memory advantage to looping over the RHS columns instead of creating the whole with=FALSE subset table?

it looks like the advantage is we can do the tmp subset once & apply it to all columns...

Reply:
I guessed that saving this tmp object and doing indexing twice is preferred over using additional memory. Rethinking it again, I might have figured out what the ToDo meant initially where using set omits this object.